File: backend/WeatherCloud_Java/src/main/java/model/WeatherData.java
package model;




public class WeatherData {

    final Location loc;
    final double temp;
    final double humidity;
    final double pressure;

    final double hdex;

    final String date;

    public WeatherData(Location loc, String date, double temp, double humidity, double pressure, double hdex){
        this.loc = loc;
        this.date = date;
        this.temp = temp;
        this.humidity = humidity;
        this.pressure = pressure;
        this.hdex = hdex;
    }

    public Location getLoc() {
        return loc;
    }

    public double getTemp() {
        return temp;
    }

    public double getHumidity() {
        return humidity;
    }

    public double getPressure() {
        return pressure;
    }

    public String getDate() {
        return date;
    }

    public double getHdex() {
        return hdex;
    }
}

File: frontend/weather_cloud_react/src/components/WeatherData.js
import React from "react";
import { useState, useEffect } from "react";
import axios from "axios";

const WeatherData = () => {
  var url = "http://192.168.10.30:8080/WeatherCloudAPI/weatherCurrent/";

  const [data, setData] = useState([]);

  const getData = async () => {
    try {
      axios.get(url).then((resp) => {
        setData(resp.data);
      });
    } catch (err) {
      console.error(err);
    }
  };

  useEffect(() => {
    getData();
    const interval = setInterval(() => {
      getData();
    }, 60000);
    return () => clearInterval(interval);
  }, []);

  console.log(data);

  const stations = [];
  data.forEach((element) => {
    stations.push(
      <div className="station">
        <h2>{element["loc"]}</h2>
        <h4>{element["date"]}</h4>
        <table>
          <tbody>
            <tr>
              <td>Temperature</td>
              <td className="val">{element["temp"]}°C</td>
            </tr>
            <tr>
              <td>Humidity</td>
              <td className="val">{element["humidity"]}%</td>
            </tr>
            <tr>
              <td>Pressure</td>
              <td className="val">{element["pressure"]}hPa</td>
            </tr>
            <tr>
              <td>Heat Index</td>
              <td className="val">{element["hdex"]}°C</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  });

  return <div className="stations">{stations}</div>;
};

export default WeatherData;
